ui/components/src/TimeSeriesTooltip/nearby-series.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ECharts as EChartsInstance } from 'echarts/core';
+import { formatValue, UnitOptions, EChartsDataFormat, OPTIMIZED_MODE_SERIES_LIMIT } from '../model';
+import { CursorData } from './tooltip-model';
+
+// increase multipliers to show more series in tooltip
+export const INCREASE_NEARBY_SERIES_MULTIPLIER = 5.5; // adjusts how many series show in tooltip (higher == more series shown)
+export const DYNAMIC_NEARBY_SERIES_MULTIPLIER = 30; // used for adjustment after series number divisor
+export const SHOW_FEWER_SERIES_LIMIT = 5;
+
+export interface NearbySeriesInfo {
+  seriesIdx: number | null;
+  datumIdx: number | null;
+  seriesName: string;
+  date: number;
+  markerColor: string;
+  x: number;
+  y: number;
+  formattedY: string;
+  isClosestToCursor: boolean;
+}
+
+export type NearbySeriesArray = NearbySeriesInfo[];
+
+/**
+ * Returns formatted series data for the points that are close to the user's cursor
+ * Adjust yBuffer to increase or decrease number of series shown
+ */
+export function checkforNearbySeries(
```

Comment:
See [other comment](https://github.com/perses/perses/pull/1181/files#r1204737458), this file was renamed, I didn't actually rewrite both of these functions.

Definitely areas to improve here, adding more early returns is one area I'm planning to look into in the future so there isn't so much nested if conditions